platform/migration/platform-migration/src/test/java/org/codice/ddf/configuration/admin/ConfigurationFilesPollerTest.java

```diff
@@ -86,362 +76,198 @@ public void setup() {
         // Setup
-        Path mockPath = getMockPath(PID, INVALID_FILE_EXT);
-        WatchEvent<?> mockWatchEvent = getMockWatchEvent(mockPath,
-                StandardWatchEventKinds.ENTRY_CREATE);
-        List<WatchEvent<?>> watchEvents = getSingleMockWatchEvent(mockWatchEvent);
-        WatchKey mockWatchKey = getMockWatchKey(watchEvents);
-        WatchService mockWatchService = getMockWatchService(mockWatchKey);
-        Path mockConfigurationDirectory = getMockConfigurationDirectoryPath(mockPath,
-                mock(Path.class));
-
-        ConfigurationFilesPoller configurationFilesPoller = createConfigurationFilesPoller(
-                mockWatchService,
-                mockConfigurationDirectory);
-
-        configurationFilesPoller.register(mockChangeListener);
+        ConfigurationFilesPoller configurationFilesPoller = new ConfigurationFilesPoller(
+                configFilePath,
+                FILE_EXT) {
+            @Override
+            void waitForFileToBeCompletelyWritten(File file) throws InterruptedException {
+                return;
+            }
+        };
+        when(configFilePath.toAbsolutePath()).thenReturn(Paths.get("foo"));
+        File file = Files.createTempFile("foo", "bar")
+                .toFile();
 
         // Perform Test
-        configurationFilesPoller.run();
+        configurationFilesPoller.register(mockChangeListener);
+        configurationFilesPoller.onDirectoryDelete(file);
 
         // Verify
-        verify(mockChangeListener, never()).notify(any(Path.class));
+        verify(mockChangeListener, never()).notify(any());
     }
 
     /**
-     * Verify that when no listener is registered, an attempt is not made to call its
-     * notify method.
+     * Verify that the ChangeListener is not notified if a RuntimeException is thrown.
      */
     @Test
-    public void testRunFileCreatedNoListenerRegistered() throws Exception {
+    public void testRuntimeExceptionThrownChangeListenerNotNotified() throws Exception {
         // Setup
-        Path mockPath = getMockPath(PID, FILE_EXT);
-        WatchEvent<?> mockWatchEvent = getMockWatchEvent(mockPath,
-                StandardWatchEventKinds.ENTRY_CREATE);
-        List<WatchEvent<?>> watchEvents = getSingleMockWatchEvent(mockWatchEvent);
-        WatchKey mockWatchKey = getMockWatchKey(watchEvents);
-        WatchService mockWatchService = getMockWatchService(mockWatchKey);
-        Path mockConfigurationDirectory = getMockConfigurationDirectoryPath(mockPath,
-                mock(Path.class));
         ConfigurationFilesPoller configurationFilesPoller = new ConfigurationFilesPoller(
-                mockConfigurationDirectory,
-                FILE_EXT,
-                mockWatchService,
-                mockExecutorService);
+                configFilePath,
+                FILE_EXT);
+        when(configFilePath.toAbsolutePath()).thenReturn(Paths.get("foo"));
+        // will cause an NPE in Paths.get()
```

Comment:
❓ Could we do `when(file.getPath()).thenThrow(new RuntimeException())` instead of relying on `Paths.get()` throwing an exception? Another option would be to overwrite `waitForFileToBeCompletelyWritten()` and have it throw the `RuntimeException`.